service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java

```diff
@@ -93,6 +108,261 @@ public void recordStewardship(
+    } else {
+      record.setTape(lastEntry.getTape());
+    }
+
+    if (updateSiteRequest.isSucker_growth() != null) {
+      record.setSuckerGrowth(updateSiteRequest.isSucker_growth());
+    } else {
+      record.setSuckerGrowth(lastEntry.getSuckerGrowth());
+    }
+
+    if (updateSiteRequest.getSite_type() != null) {
+      record.setSiteType(updateSiteRequest.getSite_type());
+    } else {
+      record.setSiteType(lastEntry.getSiteType());
+    }
+
+    if (updateSiteRequest.getSidewalk_width() != null) {
+      record.setSidewalkWidth(updateSiteRequest.getSidewalk_width());
+    } else {
+      record.setSidewalkWidth(lastEntry.getSidewalkWidth());
+    }
+
+    if (updateSiteRequest.getSite_width() != null) {
+      record.setSiteWidth(updateSiteRequest.getSite_width());
+    } else {
+      record.setSiteWidth(lastEntry.getSiteWidth());
+    }
+
+    if (updateSiteRequest.getSite_length() != null) {
+      record.setSiteLength(updateSiteRequest.getSite_length());
+    } else {
+      record.setSiteLength(lastEntry.getSiteLength());
+    }
+
+    if (updateSiteRequest.getMaterial() != null) {
+      record.setMaterial(updateSiteRequest.getMaterial());
+    } else {
+      record.setMaterial(lastEntry.getMaterial());
+    }
+
+    if (updateSiteRequest.isRaised_bed() != null) {
+      record.setRaisedBed(updateSiteRequest.isRaised_bed());
+    } else {
+      record.setRaisedBed(lastEntry.getRaisedBed());
+    }
+
+    if (updateSiteRequest.isFence() != null) {
+      record.setFence(updateSiteRequest.isFence());
+    } else {
+      record.setFence(lastEntry.getFence());
+    }
+
+    if (updateSiteRequest.isTrash() != null) {
+      record.setTrash(updateSiteRequest.isTrash());
+    } else {
+      record.setTrash(lastEntry.getTrash());
+    }
+
+    if (updateSiteRequest.isWires() != null) {
+      record.setWires(updateSiteRequest.isWires());
+    } else {
+      record.setWires(lastEntry.getWires());
+    }
+
+    if (updateSiteRequest.isGrate() != null) {
+      record.setGrate(updateSiteRequest.isGrate());
+    } else {
+      record.setGrate(lastEntry.getGrate());
+    }
+
+    if (updateSiteRequest.isStump() != null) {
+      record.setStump(updateSiteRequest.isStump());
+    } else {
+      record.setStump(lastEntry.getStump());
+    }
+
+    if (updateSiteRequest.getTree_notes() != null) {
+      record.setTreeNotes(updateSiteRequest.getTree_notes());
+    } else {
+      record.setTreeNotes(lastEntry.getTreeNotes());
+    }
+
+    if (updateSiteRequest.getSite_notes() != null) {
+      record.setSiteNotes(updateSiteRequest.getSite_notes());
+    } else {
+      record.setSiteNotes(lastEntry.getSiteNotes());
+    }
+
+    record.store();
+  }
+
+  @Override
+  public void deleteSite(JWTData userData, int siteId) {
+    isAdminCheck(userData.getPrivilegeLevel());
+    checkSiteExists(siteId);
+
+    db.deleteFrom(SITES).where(SITES.ID.eq(siteId));
+    db.deleteFrom(SITE_ENTRIES).where(SITE_ENTRIES.SITE_ID.eq(siteId));
```

Comment:
Sites has a `deleted_at` column, so all you have to do is set that to the current time. No need to actually delete the records from the database.